lib/mssql.js

```diff
@@ -476,6 +476,9 @@ MsSQL.prototype.toDatabase = function (prop, val, wrap) {
     return val;
   }
   if (prop.type.name === 'Number') {
+    if (isNaN(Number(val))) {
+      val = "'" + val +"'"; // Convert to a string so that DB will reject it
```

Comment:
Perhaps `if (isNaN(Number(val))) { val = escape(''+val); }` will do the trick?

On the other hand, I don't understand why we need to specifically handle this case, why can't we use `escape(val)` directly?